src/neo/Cryptography/Base58.cs

```diff
@@ -12,6 +12,7 @@ public static class Base58
 
         public static byte[] Base58CheckDecode(this string input)
         {
+            if (string.IsNullOrEmpty(input)) throw new ArgumentException("input cannot be null or empty string");
```

Comment:
```suggestion
            if (input is null) throw new ArgumentNullException(nameof(input));
```

Reply:
If `input` is an empty string, it should throw `FormatException`. It has been done in L17.